gamemodes/terrortown/gamemode/server/sv_player_ext.lua

```diff
@@ -789,119 +795,261 @@ local function FindCorpse(ply)
+function plymeta:Revive(delay, OnRevive, DoCheck, needsCorpse, blockRound, OnFail, spawnPos, spawnEyeAngle)
 	local ply = self
 	local name = "TTT2RevivePlayer" .. ply:EntIndex()
 
-	if timer.Exists(name) or ply.reviving then return end
+	if ply:IsReviving() then return end
 
-	if force then
-		ply.forceRevive = true
-	end
+	delay = delay or 3
+
+	ply:SetReviving(true)
+	ply:SetBlockingRevival(blockRound)
+	ply:SetRevivalStartTime(CurTime())
+	ply:SetRevivalDuration(delay)
 
-	ply.reviving = true
+	ply.OnReviveFailedCallback = OnFail
 
 	timer.Create(name, delay, 1, function()
 		if not IsValid(ply) then return end
 
-		ply.forceRevive = nil
-		ply.reviving = nil
+		ply:SetReviving(false)
+		ply:SetBlockingRevival(false)
+		ply:SendRevivalReason(nil)
 
-		if not isfunction(check) or check(ply) then
-			local corpse = FindCorpse(ply)
+		ply.OnReviveFailedCallback = nil
 
-			if needcorpse and (not IsValid(corpse) or corpse:IsOnFire()) then
-				ply:ChatPrint("You have not been revived because your body no longer exists.")
+		if not isfunction(DoCheck) or DoCheck(ply) then
+			local corpse = FindCorpse(ply)
 
-				timer.Remove(name) -- TODO needed?
+			if needsCorpse and (not IsValid(corpse) or corpse:IsOnFire()) then
+				OnReviveFailed(ply, OnFail, "message_revival_failed_missing_body")
 
 				return
 			end
 
 			ply:SpawnForRound(true)
 
-			if IsValid(corpse) then
-				local spawnPos = FindCorpsePosition(corpse)
-				if spawnPos then
-					ply:SetPos(spawnPos)
-					ply:SetEyeAngles(Angle(0, corpse:GetAngles().y, 0))
-				end
+			if not spawnPos and IsValid(corpse) then
+				spawnPos = corpse:GetPos()
+				spawnEyeAngle = Angle(0, corpse:GetAngles().y, 0)
 			end
 
-			hook.Call("PlayerLoadout", GAMEMODE, ply)
+			spawnPos = spawnPos or ply:GetDeathPosition()
+			spawnPos = spawn.MakeSpawnPointSafe(self, spawnPos)
+
+			if not spawnPos then
+				local spawnEntity = spawn.GetRandomPlayerSpawnEntity(ply)
 
+				spawnPos = spawnEntity:GetPos()
+				spawnEyeAngle = spawnEntity:EyeAngles()
+			end
+
+			ply:SetPos(spawnPos)
+			ply:SetEyeAngles(spawnEyeAngle or Angle(0, 0, 0))
 			ply:SetMaxHealth(100)
 
+			hook.Run("PlayerLoadout", ply)
+
 			local credits = CORPSE.GetCredits(corpse, 0)
 
 			ply:SetCredits(credits)
+			ply:SelectWeapon("weapon_zm_improvised")
 
 			if IsValid(corpse) then
 				corpse:Remove()
 			end
 
 			DamageLog("TTT2Revive: " .. ply:Nick() .. " has been respawned.")
 
-			if isfunction(fn) then
-				fn(ply)
+			if isfunction(OnRevive) then
+				OnRevive(ply)
 			end
-		elseif isfunction(onFail) then
-			ply:ChatPrint("Revive failed...")
-
-			onFail(ply)
+		else
+			OnReviveFailed(ply, "message_revival_failed")
 		end
 	end)
 end
 
 ---
--- Returns if a player is currently in a revival process started by @{Player:Revive}
--- @return boolean The revival status
+-- Cancel the ongoing revival process.
+-- @param [default="message_revival_canceled"]string failMessage The fail message that should be displayed for the client
```

Comment:
wrong doc style

Reply:
same here. I don't know what's wrong. This is the same style as everywhere